.github/workflows/lib-lint-and-test.yaml

```
@@ -66,21 +66,43 @@ jobs:
         uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
         with:
           persist-credentials: false
+
       - name: Set up Python
         uses: actions/setup-python@e797f83bcb11b83ae66e0230d6156d7c80228e7c # v6.0.0
         with:
-          python-version: "3.12"
-      - name: Install tox
+          python-version: "3.13"
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@3259c6206f993105e3a61b142c2d97bf4b9ef83d # v7.1.0
+        with:
+          version: "0.9.7"
+          enable-cache: false
+
+      - name: Installing dependencies
         working-directory: library
         run: |
-          pip install '.[dev]'
-      - name: Code quality checks
+          uv lock --check
+          uv sync --frozen
+
+      - name: Check formatting with ruff
         working-directory: library
-        run: tox r -vv -e pre-commit
+        run: |
+          uv run ruff check --output-format=github .
+          uv run ruff format --check .
+
+      # TODO: Disable mypy for now due to numerous typing issues that need to be resolved
+      # https://github.com/open-edge-platform/training_extensions/issues/5006
+      # - name: Check source code with mypy
+      #   working-directory: library
+      #   run: |
+      #     uv run mypy src/otx/ --config-file=pyproject.toml
```

```suggestion
      # TODO: remove 'continue-on-error' after addressing all mypy issues
      # https://github.com/open-edge-platform/training_extensions/issues/5006
      - name: Check source code with mypy
        continue-on-error: true
        working-directory: library
        run: |
          uv run mypy src/otx/ --config-file=pyproject.toml
```
On a second thought, I think it's better to keep this check enabled but allow it to fail.